### DISCUSSION TITLE
Challenge in accessing the JSON Array elements

### DISCUSSION THREAD

**Discussion by augik** [nlohmann/json#2460] [Q&A] [answered]
Dear All,
I am having difficulties accessing the json array elements. 
Here is my json file:
[{
	"File": "lane Detection",
	"Sign": [{
		"Sign1": "left_Indicator",
		"roi_x": "10",
		"roi_y": "10",
		"roi_w": "30",
		"roi_h": "30",
		"Status_all": [{
			"Status1": "enable",
			"Enable": "Yes",
			"red": "250",
			"blue": "15",
			"green": "20",
			"Tolerance": "10",
			"Frequency": "20",
			"Outcome": "135"
		}, {
			"Status2": "disable",
			"Enable": "Yes",
			"red": "50",
			"blue": "15",
			"green": "240",
			"Tolerance": "10",
			"Frequency": "20",
			"Outcome": "035"
		}, {
			"Status3": "undetected_LL",
			"Enable": "Yes",
			"red": "50",
			"blue": "15",
			"green": "20",
			"Tolerance": "10",
			"Frequency": "20",
			"Outcome": "000"
		}]
	
	}]
}]

For accessing the Status1 i have used the below command:
**for (const auto& element : j["Sign"]["Status_all"]["Status1"])
	{
		x.assign(element);
		std::cout << "The received confirmation value: " << x << std::endl;
	}**

But it is throwing the below error:
**JSON_THROW(type_error::create(305, "cannot use operator[] with a string argument with " + std::string(type_name())));**

Any kind of help to solve this issue will be highly appreciated.

Best Regards
Anayat

**Comment by gregmarr** [comment, marked as answer]
You are not doing any array indexing here.  That results in you trying to access the first element of an array using the index `"Sign"`.  You want `j[0]["Sign"]`.  

You would then be trying to access another array using the index `"Status_all"`.  You thus want `j[0]["Sign"][0]["Status_all"]`.  

Then again an array using the index `"Status1"`.  If you add a `[0]` before that to give `j[0]["Sign"][0]["Status_all"][0]["Status1"]`, then you are going to iterate over the string `"enable"`, and assigning the characters in that string to `x`.  

I'm not sure what information exactly you are trying to get out of this, and if the JSON contents are fixed, or if you can remove some of those arrays.  Hope this helps you figure out what you need to do.

**Reply by augik** [reply]
Thank you very much for your detailed answer. Now it's clear to me.
